### PR TITLE
CREATED POST REQ WAITING FOR FLASK TO DEPLOY TO HEROKU

### PR DESCRIPTION
**Pull request by CodingPenguin**
ASSIGNED LIKE A BOSS CHECK IT OUT (form.component.ts and form.service.ts probably the most important but i think it works because i tested but also the variable names are kind of confusing so maybe advice on that)

### CODE REVIEW

File: sentysis-angular/src/app/form/form.component.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      console.log('IS YOUTUBE URL');
+      this.videoId = this.sliceForVideoId(ytUrl);
+      console.log('videoId -->', this.videoId);
+      return true;
+    }
+    else {
+      console.log("NOT YOUTUBE URL: " + ytUrl);
+      return false;
+    }
+  }
```
Comment (leevanielle) [inline sentysis-angular/src/app/form/form.component.ts:62]: For each method that you write, you need to space it out with one an empty line to help make it more readable.

File: sentysis-angular/src/app/form/form.component.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      else {
+        console.log('WRONG');
+      }
+      this.youtubeForm.reset();
+    }
+    this.youtubeForm.reset();
+  }
+
+  // helper methods for onSubmit()
+  verifyUrl(ytUrl) {
```
Comment (leevanielle) [inline sentysis-angular/src/app/form/form.component.ts:50]: So what's the whole point of this method: `verifyUrl`?
Shouldn't this call be in the Flask server instead?

**Review by leevanielle — CHANGES_REQUESTED** [review]

File: sentysis-angular/src/app/form/form.component.ts (hunk 2 above)
Comment (CodingPenguin) [reply sentysis-angular/src/app/form/form.component.ts:50]: Oh this is like confirming that it's an actual youtube url, so if youtube url then proceed with the POST but if not stop the process right there

Comment (CodingPenguin) [reply sentysis-angular/src/app/form/form.component.ts:50]: Also what's the reasoning for putting it in the flask server? I thought it'd be better to do this verification in the front-end.

Comment (leevanielle) [reply sentysis-angular/src/app/form/form.component.ts:50]: Good point. Good reasoning to me! 

I recommend using Regex for this: https://stackoverflow.com/questions/19377262/regex-for-youtube-url

File: sentysis-angular/src/app/form/form.component.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    if(ytUrl.includes('https://www.youtube.com/watch?v=') || ytUrl.includes('https://youtu.be/')) {
+      console.log('IS YOUTUBE URL');
+      this.videoId = this.sliceForVideoId(ytUrl);
+      console.log('videoId -->', this.videoId);
+      return true;
+    }
+    else {
+      console.log("NOT YOUTUBE URL: " + ytUrl);
+      return false;
```
Comment (leevanielle) [inline sentysis-angular/src/app/form/form.component.ts]: In JavaScript, you can just do `return`.
Do something like this instead to warn the user that the url is faulty:
```
// Error message to let the user know 
alert('There was an error. Please make sure that the url is a youtube URL')
return;
```

File: sentysis-angular/src/app/form/form.component.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  }
+  sliceForVideoId(ytUrl) {
+    console.log ("SLICED");
+    return ytUrl.slice(-11); // ex: getVideoId(https://www.youtube.com/watch?v=abcdefghijk) returns 'abcdefghijk'
```
Comment (leevanielle) [inline sentysis-angular/src/app/form/form.component.ts]: What happens if the user put in `https://www.youtube.com/watch?v=abcdefghijklmnop`?

**Review by leevanielle — CHANGES_REQUESTED** [review]

File: sentysis-angular/src/app/form/form.component.ts (hunk 4 above)
Comment (CodingPenguin) [reply sentysis-angular/src/app/form/form.component.ts]: Oh, fixing it.

File: sentysis-angular/src/app/form/form.component.ts [hunk 5]
```diff
@@ -46,21 +45,32 @@ export class FormComponent implements OnInit {
...
-      return false;
+      alert("There was an error. Please make sure the URL is a YouTube URL.");
+      return;
     }
   }
-  sliceForVideoId(ytUrl) {
-    console.log ("SLICED");
-    return ytUrl.slice(-11); // ex: getVideoId(https://www.youtube.com/watch?v=abcdefghijk) returns 'abcdefghijk'
+
+  extractVideoId(ytUrl) {
```
Comment (leevanielle) [inline sentysis-angular/src/app/form/form.component.ts:64]: Can you explain to me what this method is doing?

File: sentysis-angular/src/app/form/form.component.ts [hunk 6]
```diff
@@ -46,21 +45,32 @@ export class FormComponent implements OnInit {
...
+    let regExp = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#\&\?]*).*/;
+    let match = ytUrl.match(regExp);
+    if (match && match[7].length == 11) {
```
Comment (leevanielle) [inline sentysis-angular/src/app/form/form.component.ts:67]: So what does this line do?

**Pull request closed without merging**